File: src/Dr.NutrizioNino.WebVue/src/Interfaces/foods/ExtractedNutrientDto.ts
export type ExtractionStatus = 'Matched' | 'IncompleteMatch' | 'Unrecognized'

export interface ExtractedNutrientDto {
  name: string
  value: number
  unit: string
  convertedValue: number | null
  canonicalUnit: string | null
  matchedNutrientId: string | null
  confidenceScore: number
  status: ExtractionStatus
}

export interface ExtractionResultDto {
  nutrients: ExtractedNutrientDto[]
  rawJson: string
}

File: Dr.NutrizioNino.Models/Dto/ExtractedNutrientDto.cs
namespace Dr.NutrizioNino.Models.Dto;

public record ExtractedNutrientDto(
    string Name,
    decimal Value,
    string Unit,
    decimal? ConvertedValue,
    string? CanonicalUnit,
    Guid? MatchedNutrientId,
    float ConfidenceScore);
